crates/hotshot/src/lib.rs

```diff
@@ -564,16 +563,9 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> SystemContext<TYPES, I> {
     }
 }
 
-/// A handle that exposes the interface that hotstuff needs to interact with a [`SystemContextInner`]
-#[derive(Clone, Debug)]
-pub struct HotShotConsensusApi<TYPES: NodeType, I: NodeImplementation<TYPES>> {
-    /// Reference to the [`SystemContextInner`]
-    pub inner: Arc<SystemContextInner<TYPES, I>>,
-}
-
 #[async_trait]
 impl<TYPES: NodeType, I: NodeImplementation<TYPES>> ConsensusApi<TYPES, I>
-    for HotShotConsensusApi<TYPES, I>
+    for SystemContext<TYPES, I>
```

Comment:
I know this is still a draft, but I think we want the handle to impl this trait so we can avoid the whole `c_api = ...` thing.  

Also maybe we want to get rid of the api trait altogether.  Would we ever have another implementation of `ConsensusApi`?